### ISSUE TITLE
Mixer design express each instrument's volume as relative to whole instead of to itself

### ISSUE THREAD

**Issue by ericfont** [musescore/MuseScore_design#9]
I'm guessing this repo is where people are allowed to discuss/brainstorm 3.0 design ideas...I came from https://musescore.org/en/node/188266 which asks the question:
  "2. As I am interacting with a computer, what can it do for me that traditional equipment can't?"

Currently the mixer has a gain knob for each instrument's dB telling synth how much to bring up or down that part in the mix relative to the *default level of its soundfont*.  But it think it would be useful to represent each instrument's volume as relative to the entire volume.  (Keep in mind, the audio export will always normalize the waveform, and keep in mind there is a separate master volume level, so mixer is more appropriate for relative levels.)

I've done a mockup on the right of how the array of volume and panning knobs as musescore's current mixer would correspond to a 2-D representation of each instrument's relative volume to whole:

![mixer-mockup](https://cloud.githubusercontent.com/assets/6502474/25304592/71aa5cf8-2738-11e7-9f16-f73ed257438d.png)

Notice that each instrument's left edge represents the relative volume of it's left channel component relative to the total volume of left channel, while each rightmost edge represents that instrument's right component relative to the total volume of the right channel.  (The percentage values on the left and right reflect that).

Immediately the user is able visually see roughly how loud each instrument is relative to the whole, and is able to see how far left or right they are.  Both of these things are something which the traditional mixer board does not communicate well.

I can imagine that the user could manipulate this 2-D map with the mouse.  For one, if used the middle mouse wheel over an enclosed area, scrolling up could increase the relative size of the area (which would necessarily simultaneously decrease the area of every other instrument), and vice versa for scrolling down.  I also imagine that left-clicking inside each region could also be a way to increase it's relative size (and right-
 clicking could decrease relative size).  (And even maybe user could drag each line up or down, but I suppose that wouldn't be terribly useful since it would change the volume for two neighboring instruments, which may not be desired.)  Hypothetically to adjust the panning, user could click and drag an instrument's label and move it left or right...and in real time the size of the that instrument's left and right edge would adjust their length to reflect the new panning.

Oh, I also colored instruments according by their type.  I think this is useful visual feedback.  I would think all instruments of the same instrument family should be similarly colored, just with slightly different shades.

**Comment by ericfont** [comment]
(There is also another thing I should bring up...I have used the word loud above, but there is a distinction between loudness and amplitude, which I am aware of.  Hypothetically, I do think it would be useful when loading a soundfont to calculate its "loudness" as well...since different soundfonts may sound louder or softer even if they had identical level on the knob.)

**Comment by ghost** [comment]
You said,
_And even maybe user could drag each line up or down..._
Would cause the adjacent instrument's volume to change. It would be possible to change all other instruments volume in relationship to this just as spinning the dial currently does.

**Comment by ericfont** [comment]
The problem I thought I had with having dragging a line change all other instruments volume is: how do I know which instrument I am increasing...the one above or below the line?

But now that I think about it, I was actually only imagining dragging the non-vertifcal lines...But now I'm thinking some more and I could imagine it would work for dragging the vertical lines if have move up mean to increase that instrument's left or right channel volume, while dragging down would decrease its relative volume.

It also occurs to me that dragging label up/down could increase/decrease that instruments volume.

It also occured  I to me now I that dont neceasarily have to drag the label or the vertical lines, but instead could just drag from anywhere within the area, although the left and right vertical lines would be specifically for left/right volume.  Similarly could left click each line to increase and right-click to decrease

**Comment by ericfont** [comment]
I just had an idea that user could select multiple instreumts, either by dragging a section box around all of their labels, or by Shift-Clicking on their areas, and could then drag up/down to increase/decrease all their volume relative to whole.  Maybe even drag left/right for panning.

**Comment by ericfont** [comment]
It occured to me know that maybe I should have all areas be vertically symmetric only, and then for instruments that are panned there will be whitespace above and below them, so won't worry if their areas touch the instrument above or below them.  That also helps with drawing their labels horizontally only.  I can make a better mockup later if this is not clear.

**Comment by MikelNelson** [comment]
Nice work, Eric.

In your example the Trombones 1&2 just happen to pan so that the sum of the volumes is constant, so the bottom and top lines can lie together. But there may be an instrument that has no single compensating instrument. Or indeed the total volume may be different for left and right. Obviously you know this, and it seemed like this is what you were referring to in your last comment about whitespace, but I couldn't understand it properly, so a mock-up would be good!

**Comment by ericfont** [comment]
You bring up a good point.  I did eventually come to consider that problem.

I have actually spent a few hours, and eventually came up with what I believe is a much better design #10, so I'm closing this in favor of that.  The biggest problems with this design (#9) are:
1. Cannot be easily integrated into the mixer, since some areas will be too large and some will be too small.
2. Will have difficulties including instrument labels if shape is too small or too slanted to fit instrument label
3. The point you mentioned about total volume being different for Left & Right channels.
4. The fact that signals panned to the center actually have their left and right channel gains attenuated by 3dB...this is a minor technical detail, but still I don't like things being incorrect.
5. Not terribly intuitive to a new users
6. a few other things...

Anyway, I think my new design #10 is much improved and addresses all these issues.  But thinking about this here and then sleeping on it was necessary in order to come up with my new better design.
